23 sprint

I1kham opened the pull request.

I1kham committed:
sprint 23
added icon
clean unused res

I1kham committed:
sprint 23
added icon
clean unused res

I1kham committed:
sprint 23
added icon
clean unused res

I1kham committed:
sprint 23
добавлен экран редактирования плейлиста
внедрен фрагмент в указанный экран для последующей его замены на соответствующий

I1kham committed:
sprint 23
добавлен экран редактирования плейлиста
внедрен фрагмент в указанный экран для последующей его замены на соответствующий

I1kham committed:
sprint 23
добавлен экран редактирования плейлиста
внедрен фрагмент в указанный экран для последующей его замены на соответствующий

I1kham committed:
sprint 23
добавлен экран редактирования плейлиста
внедрен фрагмент в указанный экран для последующей его замены на соответствующий

I1kham committed:
sprint 23
добавлен экран редактирования плейлиста
внедрен фрагмент в указанный экран для последующей его замены на соответствующий

I1kham committed:
sprint 23
добавлен экран редактирования плейлиста
внедрен фрагмент в указанный экран для последующей его замены на соответствующий

I1kham committed:
sprint 23

I1kham committed:
sprint 23

I1kham committed:
sprint 23

I1kham committed:
sprint 23

I1kham committed:
sprint 23

I1kham committed:
sprint 23

I1kham committed:
sprint 23

I1kham committed:
sprint 23

I1kham committed:
sprint 23

I1kham committed:
sprint 23

I1kham committed:
sprint 23

I1kham committed:
sprint 23

I1kham committed:
Revert "sprint 23"

This reverts commit 8fc266ebf4a09dc2d68cf0d6c733f7ac8f36542b.

I1kham committed:
sprint 23

I1kham committed:
sprint 23

I1kham committed:
sprint 23

I1kham committed:
sprint 23

I1kham committed:
sprint 23

I1kham committed:
sprint 23

I1kham committed:
sprint 23

I1kham committed:
sprint 23

kosstyango commented:
⚠️ Критические замечания:
При добавлении трека в плейлист всплывающее окно не убирается.
Треки в плейлисте должны отображаться в обратном порядке: последний добавленный должен отображаться вверху.
При нажатии на трек в результатах поиска приложение "падает" в 10% случаев. (Скрин ошибок)
![Спринт_23_Хабибуллин](https://github.com/user-attachments/assets/c6e3ebfa-eb10-4a64-a04b-a37ce26e2cc4)

При отправке плейлиста по кнопке "Поделиться" в сообщении два раза повторяется цифра с количеством треков в плейлисте: "2 2 трека", "5 5 треков" итд.
При редактировании плейлиста невозможно изменить картинку и поля "Название" и "Описание", а кнопка "Сохранить" всегда находится в неактивном состоянии.

I1kham committed:
sprint 23
исправлена проблема с отображением списка, реверс по добавдлению

I1kham committed:
sprint 23
проверка проблемы с всплывающим сообщением
не подтвердилась, добавлена возможность скрыть сообщение по клику

I1kham committed:
sprint 23
исправлена проблема с крашем приложения при тапе  на эеране поиска треков на трек

I1kham committed:
sprint 23
исправлен баг с сохранением картинок и отображением их в разных окнах приложения

I1kham committed:
sprint 23
исправлен баг с сохранением картинок и отображением их в разных окнах приложения

I1kham committed:
sprint 23
исправлена ошибка с отправкой плейлиста

I1kham committed:
sprint 23
исправлена ошибка с отправкой плейлиста

I1kham commented:
привет @kosstyango 

пробежался по замечаниям:

1. При добавлении трека в плейлист всплывающее окно не убирается.
не совсем понял, какое окно имеется ввиду. Если окно со списком плейлистов, то не видел указаний, что оно должно было убираться по нажатию на элемент списка. А если имеется ввиду окно с сообщением, что трек добавлен, то это окно должно убираться автоматически, но я добавил возможность скрыть сообщение не только смахнув вниз, но и кликом

2. Треки в плейлисте должны отображаться в обратном порядке: последний добавленный должен отображаться вверху.
да, я использовал HashSet, для удобства реализации, не проверять на наличие трека в списке, но HashSet еще и сортирует, поэтому порядок нарушался, исправлено

3. При нажатии на трек в результатах поиска приложение "падает" в 10% случаев. (Скрин ошибок)
да, причину нашел, переход на экран плеера происходил быстрее, чем room успевал записать в базу новый трек, сделал вызов перехода после окончания записи. исправлено

4. При отправке плейлиста по кнопке "Поделиться" в сообщении два раза повторяется цифра с количеством треков в плейлисте: "2 2 трека", "5 5 треков" итд. 
исправлено

5. При редактировании плейлиста невозможно изменить картинку и поля "Название" и "Описание", а кнопка "Сохранить" всегда находится в неактивном состоянии.
вот тут поподробнее, да я забыл включить логику замены картинки плейлиста, то все textView были кликабельны и редактируемы. Да логика экрана редактирования была чуть другая, я реализовывал логику, что если не внесены изменения в информацию, то и сохранять нечего

если данное поведение повторится, можно чуть больше подробностей, у меня оно не проявилось, пробовал на виртуалках и физических устройствах на всех api с 29 по 34

I1kham committed:
sprint 23
при успешном добавлении трека в пелйлисть на эеране плеера, список плейлистов скрывается, а пользователь видит сообщение, что трек успешно добавлен.
 есди трек не добавлен, то список плейлистов остается видимым и есть сообщение, что в плейлисте трек уже содержится

I1kham committed:
sprint 23
проверка и приведение эекрана создания плейлиста к заданию 22 спринта

I1kham committed:
sprint 23
проверка экрана добавления и редактирования плейлиста на соответсвие 23 спринта, соответствует,
изменение сообщения со списком треков

исправлен бага с сохранением изменений плейлиста, в частности обложка плейлиста удалялась до проверки ее наличия в базе
